src/components/Calendar/DayCard/DayCard.js

```diff
@@ -0,0 +1,222 @@
+    }
+  }
+`;
+
+const StyledSide = styled.div`
+  position: absolute;
+  width: 100%;
+  height: 100%;
+  -webkit-backface-visibility: hidden;
+  backface-visibility: hidden;
+  border-radius: 10px;
+  cursor: pointer;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  font-size: 3rem;
+`;
+
+const StyledFrontSide = styled(StyledSide)`
+  background-color: powderblue;
+  color: black;
+
+  & > span {
+    bottom: 3px;
+    position: absolute;
+    font-size: 2rem;
+  }
+`;
+
+const StyledBackSide = styled(StyledSide)`
+  color: white;
+  transform: rotateY(180deg);
+`;
+
+// const StyledModal = styled.div`
+//   position: absolute;
+//   background: powderblue;
+//   z-index: 20;
+//   border-radius: 10px;
+//   display: ${({ isModalVisible }) => (isModalVisible ? 'block' : 'none')};
+// `;
+
+const StyledModalBackground = styled.div`
+  position: fixed;
+  width: 100%;
+  height: 100%;
+  left: -20px;
+  top: -20px;
+  display: ${({ isModalVisible }) => (isModalVisible ? 'block' : 'none')};
+  z-index: 2;
+`;
+
+// const Modal = ({ isModalVisible }) => {
+//   return <StyledModal isModalVisible={isModalVisible}></StyledModal>;
+// };
+
+const DayCard = ({ front, back, className }) => {
+  const card = useRef(null);
+  const modal = useRef(null);
+
+  const [isRevealed, setRevealed] = useState(0);
+  const [isModalVisible, setModalVisible] = useState(0);
+  const [coordinates, setCoordinates] = useState();
+
+  const handleReveal = () => {
+    setRevealed(1);
+  };
+
+  const showDayModal = () => {
+    if (isRevealed) {
+      card.current.style.opacity = 0;
+
+      if (!isModalVisible) {
+        setCoordinates(card.current.getBoundingClientRect());
+        const { top, left, height, width } = coordinates ? coordinates : card.current.getBoundingClientRect();
+
+        gsap.fromTo(
+          modal.current,
+          { top, left, width, height },
+          {
+            duration: 0.8,
+            top: `calc(50% - ${450 / 2}px)`,
+            left: `calc(50% - ${400 / 2}px)`,
+            width: '400px',
+            height: '450px',
+          }
+        );
+
+        setModalVisible(1);
+      } else {
+        card.current.style.opacity = 1;
+        const { top, left, height, width } = coordinates;
+
+        gsap.to(modal.current, { duration: 0.8, top, left, width, height, opacity: 0 });
+
+        setModalVisible(0);
+      }
+    }
+  };
+
+  const handleClick = () => {
+    handleReveal();
+    showDayModal();
+  };
+
+  return (
+    <>
+      <StyledDayCardContainer
+        ref={card}
+        className={className}
+        isRevealed={isRevealed}
+        onClick={() => handleClick(modal)}
+      >
+        <StyledCard>
+          <StyledFrontSide>
+            {front}
+            <span>29.11.20</span>
+          </StyledFrontSide>
+          <StyledBackSide>{back}</StyledBackSide>
+        </StyledCard>
+      </StyledDayCardContainer>
+      <StyledModal ref={modal} isModalVisible={isModalVisible}>
```

Comment:
If it comes to modal case and if you're not satisfied with 1 modal / card approach - wdyt about creating one modal a little higher into the component tree + creating context that will have a method, named for example `showModal`, that will take coordinates of current card (`card.current.getBoundingClientRect()`) and some component to render inside. When that method is called, we could perform the animation, knowing the initial coordinates. Based on that we could split `Modal` (that could be a little more generic) from `DayCard`